examples/tree_2d_dgsem/elixir_kpp.jl

```diff
@@ -0,0 +1,152 @@
+using Trixi
+using OrdinaryDiffEq
+using Plots
+using LinearAlgebra
+
+###############################################################################
+# Definition of the 2D scalar "KPP equation"
+#
+# See: Kurganov, A., Petrova, G., and Popov, B. (2007).
+# Adaptive Semidiscrete Central-Upwind Schemes for Nonconvex Hyperbolic Conservation Laws,
+# SIAM Journal on Scientific Computing, 29(6), 2381--2401
+
+struct KPPEquation2D <: Trixi.AbstractEquations{2, 1} end
+
+# The KPP flux is F(u) = (sin(u), cos(u))
+@inline function Trixi.flux(u, orientation::Integer, ::KPPEquation2D)
+    if orientation == 1
+        return SVector(sin(u[1]))
+    else
+        return SVector(cos(u[1]))
+    end
+end
+
+# Since the KPP problem is a scalar equation, the entropy-conservative flux is uniquely determined
+@inline function Trixi.flux_ec(u_ll, u_rr, orientation::Integer, ::KPPEquation2D)
+    if abs(u_ll[1] - u_rr[1]) < 1e-12
+        return 0.5*(flux(u_ll, orientation, KPPEquation2D())+flux(u_rr, orientation, KPPEquation2D()))
+    else
+        factor = 1.0/(u_rr[1] - u_ll[1])
+        if orientation == 1
+            return SVector(factor*(-cos(u_rr[1]) + cos(u_ll[1])))
+        else
+            return SVector(factor*(sin(u_rr[1]) - sin(u_ll[1])))
+        end
+    end
+end
+
+# Wavespeeds
+const kpp_wavespeed = 1.0
+@inline Trixi.max_abs_speeds(u, ::KPPEquation2D) = (kpp_wavespeed, kpp_wavespeed)
```

Comment:
Can `kpp_wavespeed` be stored in `KPPEquation2D` instead? You could unpack it via 
```julia
@inline Trixi.max_abs_speeds(u, equations::KPPEquation2D) = (equations.kpp_wavespeed, equations.kpp_wavespeed)
```

Reply:
Or just avoid storing the global constant and use something like the following?
```suggestion
@inline function Trixi.max_abs_speeds(u, ::KPPEquation2D)
  # Rough estimate of the maximal wave speed from above
  kpp_wavespeed = 1.0
  return (kpp_wavespeed, kpp_wavespeed)
end
```

Reply:
In that case, what should be done for `max_abs_speed_naive`?

Reply:
Something along the same lines? As @jlchan said, I would just like to avoid using a global variable

Reply:
I'm happy to make whatever change you recommend, but I am having trouble seeing the benefit.

Introducing a named local variable in `max_abs_speeds` is not ideal, because the same constant should be used also in `max_abs_speed_naive`, and so it should live in some broader scope.

If, on the other hand, the wavespeed is a field of the `KPPEquation2D` struct, it cannot be treated as a constant. I doubt this has any practical implications, but in principle a constant seems better?

Reply:
> If, on the other hand, the wavespeed is a field of the KPPEquation2D struct, it cannot be treated as a constant. I doubt this has any practical implications, but in principle a constant seems better? 

I don't think I understand why you're looking for a constant - are you looking to make the constant globally accessible? 

> Yet another option would be to define a function like @inline wavespeed(::KPPEquation2D) = 1.0. This seems semantically pretty similar to a global constant

This seems good to me.

Reply:
Using a constant will allow for constant propagation, so the compiler can do more computations at compile time rather than run time. I think in this particular instance it will not make any practical difference, so it's probably a moot point. Semantically it also makes sense to represent such a value as a constant rather than as a "variable that's not really variable".

I will go with the function approach (unless @ranocha objects). It's similar in a lot of ways to using a constant (it should also allow for constant propagation), but maybe is a bit better behaved in terms of namespacing, and might be more acceptable if there's a general aversion to using globals.

Reply:
Sounds good to me, too.